qiime2/sdk/results.py

```diff
@@ -44,14 +45,22 @@ def __new__(cls, fields, values):
 
         # Must set attributes this way because `__setattr__` prevents
         # setting directly (necessary for immutability).
-        object.__setattr__(self, '_fields', fields)
+        object.__setattr__(self, "_fields", fields)
 
         # Attach field names as instance attributes.
         for field, value in zip(fields, values):
             object.__setattr__(self, field, value)
 
         return self
 
+    def __iter__(self):
```

Comment:
This breaks our API almost everywhere. The default `__iter__` of tuple permits unpacking of results:
```python
a, b = resultsObject
```